docs/deployment/private/self-hosted-deployment.mdx

```diff
@@ -0,0 +1,272 @@
+
+Next, you will need to access the `.env.cloud` file which gets generated at the root of your supergraph.
+
+For each READ_URL & WRITE_URL, you will need to make the necessary adjustments to ensure that your v3-engine (Which will be running in your Kubernetes cluster) is properly configured to reach each one of your connectors.  The current values that these environment variables point to are for local connectivity.
+
+The URLs of these environment variables needs to be structured as shown below.
+
+:::info
+
+
+URL: **http://`<connector-helm-release-name>`-`<connector-helm-chart-name>`.`<namespace>`:8080**
+
+- `<connector-helm-release-name>`: Helm release name that you will be using later on during your connector deployments.
+
+- `<connector-helm-chart-name>`: Helm chart name of your connector.
+
+- `<namespace>`: Namespace where your connector will be deployed to.
+:::
+
+After making the necessary changes to your `.env.cloud` file, run the below command.  This will create a cloud build and will also generate the necessary artifacts which will later on be consumed by your v3-engine. 
+
+```bash title="Create a build for your cloud project."
+ddn supergraph build create --self-hosted-data-plane --output-dir build-data --project <ddn-project-name> --out json
+```
+
+At this point, take note of the `<build_version>` and `<observability_hostname>` which will be outputted here.  You will need these later.
+
+## Step 3. Build image for v3-engine
+
+Ensure that you are running the commands from the root of your supergraph.
+
+```bash title="Create a Dockerfile for v3-engine."
+cat <<EOF >> Dockerfile
+FROM ghcr.io/hasura/v3-engine
+COPY ./build-data /md/
+EOF
+```
+
+:::warning Building images with proper OS/Arch
+
+Ensure that you are building the image with the correct OS/arch which would enable the image to run properly under your Kubernetes cluster
+:::
+
+```bash title="Build the image via docker build.  Tag this image with your own registry and a custom tag of your choosing."
+docker build -t <your_registry>/v3-engine:<your_tag> .
+```
+
+```bash title="Push the image to your registry."
+docker push <your_registry>/v3-engine:<your_tag>
+```
+
+## Step 4. Build image for each connector
+
+:::warning Building images with proper OS/Arch
+
+Ensure that you are building the image with the correct OS/arch which would enable the image to run properly under your Kubernetes cluster
+:::
+
+```bash title="Build image via docker compose.  Find the <service-name> from app/connector/<connector>/compose.yaml"
+docker compose build <service-name>
+```
+
+```bash title="Re-tag the image"
+docker tag <root-folder-name>-app_<connector> <your_registry>/<connector>:<your_tag>
+```
+
+```bash title="Push the image to your registry."
+docker push <your_registry>/<connector>:<your_tag>
+```
+
+## Step 5. Create ingresses on your Kubernetes cluster
```

Comment:
How do we create ingress without deploying the engine service? I think we should move this to a step after deploying via Helm

Reply:
Yup, will make the change!